### PR TITLE
feat: add basic SentryCli type support

### PR DESCRIPTION
**Pull request by ssthouse**
hi，I add the basic SentryCli type definitions and test locally with `npm link`.

the result looks fine.

image

image

### CODE REVIEW

**Comment by ssthouse** [comment]
#560

**Comment by ssthouse** [comment]
@kamilogorek hi, please check when you have time : D

File: js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+declare module '@sentry/cli' {
+  export interface SentryCliOptions {
+    url?: string;
+    authToken?: string;
+    apiKey?: string;
+    dsn?: string;
+    org?: string;
+    project?: string;
+    vscRemote?: string;
+    silent: boolean;
```
Comment (kamilogorek) [inline js/index.d.ts:13]: `silent` can be marked optional, as we only change the behavior if it's `true`. Any falsy value will work for disabling it.
```suggestion
    silent?: boolean;
```

File: js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  export default class SentryCli {
+    constructor(configFile?: string | null, options?: SentryCliOptions)
+
+    public configFile?: string;
+    public options?: SentryCliOptions;
+    public releases: SentryCliReleases
+
+    public static getVersion(): string
+    public static getPath(): string
+    public static execute(args: string[], live: boolean): Promise<string>
```
Comment (kamilogorek) [inline js/index.d.ts:85]: `execute` is not static in this case:

```suggestion
    public execute(args: string[], live: boolean): Promise<string>
```

File: js/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+  export interface SentryCliCommitsOptions {
+    auto?: boolean;
+    repo?: string;
+    commit?: string;
+    previousCommit?: string;
+  }
+
+  export interface SentryCliReleases {
+    ['new'](
```
Comment (kamilogorek) [inline js/index.d.ts:47]: is `new` reserved keyword here and we cannot use `new(release: string...`? 😶

**Review by kamilogorek — CHANGES_REQUESTED** [review]
Just a few details. Thanks, really appreciate it :)

File: js/index.d.ts (hunk 3 above)
Comment (ssthouse) [reply js/index.d.ts:47]: yes~

**Review by kamilogorek — APPROVED** [review]

**Comment by kamilogorek** [comment]
Thanks! Will try to release it this week.

**Comment by ckknight** [comment]
@kamilogorek I'm noticing that the `.d.ts` file isn't being included in the package published on npm (v1.63.2)

**Comment by kamilogorek** [comment]
@ckknight it's because this commit has not been release yet :) I'll cut off a new release someday this week.

**Comment by kamilogorek** [comment]
@ckknight release has been cut off. Should be deployed any hour tomorrow as `1.64.0`. Cheers!

**Pull request merged**